### ISSUE TITLE
PermanentlyClearPreviousMailboxInfo switch not working?

### ISSUE THREAD

**Issue by jonwbstr** [MicrosoftDocs/office-docs-powershell#7028]
Hello,

This article says that using the "PermanentlyClearPreviousMailboxInfo" switch should remove info from the mailbox that will let us reset the mailbox. I have removed license from mailbox, ran the command and reassigned a license. The previously detached mailbox with items was reattached to the user, when the goal is to get a NEW, empty mailbox.

https://techcommunity.microsoft.com/t5/exchange-team-blog/permanently-clear-previous-mailbox-info/ba-p/607619

I'm thinking this switch is no longer supported (it really should be), and I'll need to use the traditional method of deleting the user and purging the user to get the exchangeGuid cleared successfully ;(

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9ae09acf-5413-eb3c-e4c7-2481b7b178f5
* Version Independent ID: d0d555a7-de13-0173-7a8c-e13587b334bc
* Content: [Set-User (ExchangePowerShell)](https://docs.microsoft.com/en-us/powershell/module/exchange/set-user?view=exchange-ps)
* Content Source: [exchange/exchange-ps/exchange/Set-User.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/exchange/exchange-ps/exchange/Set-User.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by dariomws** [comment]
@jonwbstr - thank you for submitting feedback.

It seems that the parameter still works. In this case, I think the best way forward is if you open a service ticket in your tenant so this can get resolved ASAP. Based on the outcome, let me know if it can be called out in the docs.

Please follow this link to contact support for business products:  https://docs.microsoft.com/office365/admin/contact-support-for-business-products.

Please keep us posted here on the resolution to feed whatever information you discover into the content.

**Comment by jonwbstr** [comment]
Really? I've had 3 people from ms support reproduce the issue, what version
of the module did you use?

**Comment by jonwbstr** [comment]
Sorry 2 people, I'm the 3rd 😁

**Comment by dariomws** [comment]
If you have a Microsoft alias, you can link your accounts in GitHub so that a blue "Microsoft submitter" label appears to any PR's or issue you submit. Guidance on it here: https://docs.opensource.microsoft.com/tools/github/index.html

Here is the site you can use to see the mappings between GitHub and Microsoft aliases: https://repos.opensource.microsoft.com/people?q=jonwbstr

Could you kindly confirm this information with the product team before we edit the article?

**Comment by dariomws** [comment]
@jonwbstr I don't have a hybrid Exchange lab to test it properly, but I used it in November 2020. Are you trying with the EXO PS V2 2.0.3, right? I'm trying to confirm it on my own, but I would also appreciate it if you can try it.

Thanks!

**Comment by jonwbstr** [comment]
Both v2 and the basic auth method are  giving the same unexpected behavior.
I'm not a Microsoft employee, but I did sleep at a Holiday Inn last night.
And I'm a CSP partner.

I've never tried this but could probably spin up a demo and test there if
needed.
https://cdx.transform.microsoft.com/

Support has tech leads looking into the report at the moment

-jon

**Comment by dariomws** [comment]
> Support has tech leads looking into the report at the moment

Please keep us posted here on the resolution to feed whatever information you discover into the content.

**Comment by jonwbstr** [comment]
Waiting on PG to review the case

**Comment by jonwbstr** [comment]
you mention hybrid exchange. So I wonder if the switch is not clearing the properties the docs say it is clearing.
Article says (ps, the CAUTION tag is broken)

> This switch permanently deletes the existing cloud mailbox and its associated archive, prevents you from reconnecting to the mailbox, and prevents you from recovering content from the mailbox.

I am not trying to use this switch in a hybrid environment, as written this command should work on a cloud only mailbox. I don't want to delete the user and reconfigure everything just to reset the user's mailbox and archive. They have group memberships and policies applied to them... They are AADConnect so deleting the user and recreating them is an involved process.

If the switch would "permanently deletes the existing cloud mailbox and its associated archive" it meets my needs perfectly.

**Comment by jonwbstr** [comment]
To verify,

You confirmed it works, and we still don't know why it's not working as
intended for me in this non hybrid scenario.

Is that accurate?

**Comment by dariomws** [comment]
@jonwbstr,

Yes, I confirmed it works.

The best way forward in your scenario is if you open a service ticket in your tenant so this can get resolved ASAP. Based on the outcome, let me know if it can be called out in the docs.

Please follow this link to contact support for business products:  https://docs.microsoft.com/office365/admin/contact-support-for-business-products.

Please keep us posted here on the resolution to feed whatever information you discover into the content.

**Comment by yogkumgit** [comment]
@dariomws  Thank you very much for the contribution and sharing this explanation. @jonwbstr Hope this comment is helpful for you. Considering the nature of the issue, as advised, please open a service ticket in your tenant and follow with them for the resolution. We proceed here to close it. 
We will be very happy if you can share the outcome or resolution with us if you see documentation update is required. We will certainly update this documentation with that valuable information. Thanks again, for taking out some time to open the issue. Appreciate and encourage you to do the same in future also.

**Comment by jonwbstr** [comment]
Okay, I will open a new issue when I have an update since you want to close
this before the ticket I opened 2 weeks ago is still researching this issue
And your in a hurry to close the issue with the documentation.

Maybe you could help move my ticket along instead of sweeping it under the
rug?

**Comment by jonwbstr** [comment]
Case #:23544071

I have no trouble reproducing the issue. We have someone saying it works.
I've told you my scenario where it doesn't work.

Can you elaborate on the scenario where it works? So we can help your
colleagues in the support team with this issue. They seem to be having
trouble figuring out why it doesn't work in my scenario.


It's great for the doc to suggest this is helpful in hybrid scenarios but
no where does it say a hybrid scenario is the only situation where it works.

So what's up?

**Comment by dariomws** [comment]
Hi @jonwbstr, thank you for your comment.

Here we can only update the article, and we cannot support you, my apologies. The article states that this parameter might be required in mailbox move and re-licensing scenarios between on-premises Exchange and Microsoft 365. And it provides an article with more information. The parameter was added more than 2 years ago with that in mind, so it's not intended to use in other scenarios.
I used it 4 months ago in a hybrid scenario, and it worked. Asked some more colleagues and then confirmed me that they used it recently also in a hybrid scenario.

I suggest that you talk with support to see if there is another way to do what you need.

**Comment by jonwbstr** [comment]
Dario,

Here's the thing. The article doesn't say it only works in hybrid scenarios. it says it can be used in hybrid scenarios. I don't know. I've reported doc bugs where the doc maintainer takes it upon themselves to get clarification from the PG and collogues and updates the article based on new information collected on their own. Other maintainers expect to the reporter who is saying the documentation is incorrect, to open a support ticket hunt down the updates the maintainer needs, and then the maintainer might update the article. They might also choose to claim no updates are needed because of an obscure reference that COULD be interrupted to mean it's working as intended when clearly someone was confused because they opened the ticket in the first place.

As I've said, as written the article is not accurate. You have chosen to close the issue suggesting you believe it is accurate and now I have to prove you wrong because you are more interested in closing the issue and less interested in improving the accuracy of the article. Why should I care if your document is up to date I should just open support tickets every time I have an issue and skip the documentation maybe support will care about the article getting fixed.
Why? Because I care about improving the experience for my fellow o365 administrators.

Hopefully you will care about improving the experience as much as I do one day!

**Comment by chrisda** [comment]
@jonwbstr, please do not question how much someone cares. I've worked with Dario for the last couple of years, and trust me, he cares. His very first response to your issue was the suggestion to file a support case, which you did, and which is 100% the best course of action for something like this. This puts the problem front and center in front of the product team and demands an answer. The problem is, you haven't received an answer yet, no? That will ultimately dictate what we do to the topic. Who knows, you might have uncovered a bug where your scenario is supposed to work, but it doesn't for whatever reason?

We're all eager to hear the results of your support case. I'm reopening the issue.

**Comment by chrisda** [comment]
@jonwbstr, anything new to report?

**Comment by jonwbstr** [comment]
Support is still going around in circles. The texas power outage didn't
help. I asked for an escalation this morning

**Comment by jonwbstr** [comment]
Spoke with person assigned to my case today, and someone is working on updating the documentation. The switch is not supported in Exchange Online.

**Comment by chrisda** [comment]
@jonwbstr, it would appear no one is working to update the documentation, or it would have been done by now. So, I take it that the solution is to say that parameter doesn't work in Exchange Online?

**Comment by jonwbstr** [comment]
I've got an email from 3/15 where one of the case owners asked for the docbug issue # and nobody replied. I don't see any signs the case was closed so l pinged everyone in the ticket asking for an update. It might take a few days to run this up the flag pole so to speak.

**Comment by yogkumgit** [comment]
@jonwbstr

**Comment by jonwbstr** [comment]
Hmm, thought I updated this last month. It was one of the typical we are still researching type updates. I'm going to escalate.

**Comment by jonwbstr** [comment]
I put @chrisda and our ASfP account manager in touch with each other today. They should be able to track down someone related to the support ticket and get this sorted out. I thought the answer was, "the switch is not supported" but instead got an "we will update the documentation" so I'm still not completely clear on what the correct answer is. :)

**Comment by chrisda** [comment]
I'm updating the parameter description in the topic with standard boilerplate for "available in this environment but doesn't work":

> This parameter is available or functional only in on-premise Exchange

If this isn't quite right, someone else can tweak as appropriate.

Thanks for the Feedback.

(Finally) Closing
